packages/sanity/src/core/studio/addOnDataset/setup/CommentsSetupProvider.tsx

```diff
@@ -51,60 +54,56 @@ export function CommentsSetupProvider(props: CommentsSetupProviderProps) {
     [originalClient, projectId],
   )
 
-  const handleRunSetup = useCallback(
-    async (comment: CommentPostPayload) => {
-      setIsRunningSetup(true)
-
-      // Before running the setup, we check if the addon dataset already exists.
-      // The addon dataset might already exist if another user has already run
-      // the setup, but the current user has not refreshed the page yet and
-      // therefore don't have a client for the addon dataset yet.
-      try {
-        const addonDatasetName = await getAddonDatasetName()
-
-        if (addonDatasetName) {
-          const client = handleCreateClient(addonDatasetName)
-          setAddonDatasetClient(client)
-          await client.create(comment)
-          setIsRunningSetup(false)
-          return
-        }
-      } catch (_) {
-        // If the dataset does not exist we will get an error, but we can ignore
-        // it since we will create the dataset in the next step.
-      }
+  const handleRunSetup = useCallback(async (): Promise<SanityClient | null> => {
+    setIsRunningSetup(true)
+
+    // Before running the setup, we check if the addon dataset already exists.
+    // The addon dataset might already exist if another user has already run
+    // the setup, but the current user has not refreshed the page yet and
+    // therefore don't have a client for the addon dataset yet.
+    try {
+      const addonDatasetName = await getAddonDatasetName()
 
-      try {
-        // 1. Create the addon dataset
-        const res = await originalClient.withConfig({apiVersion: API_VERSION}).request({
-          uri: `/comments/${dataset}/setup`,
-          method: 'POST',
-        })
-
-        const datasetName = res?.datasetName
-
-        // 2. We can't continue if the addon dataset name is not returned
-        if (!datasetName) {
-          setIsRunningSetup(false)
-          return
-        }
-
-        // 3. Create a client for the addon dataset and set it in the context value
-        //    so that the consumers can use it to execute comment operations and set up
-        //    the real time listener for the addon dataset.
-        const client = handleCreateClient(datasetName)
+      if (addonDatasetName) {
+        const client = handleCreateClient(addonDatasetName)
         setAddonDatasetClient(client)
+        setIsRunningSetup(false)
+        return client
+      }
+    } catch (_) {
```

Comment:
I'm afraid I'm not as close to this as you are, but is there a specific error we can catch here? I'm just concerned about valid errors being ignored and further work being done, getting into a strange state.

The additional error catching below might suffice, but was curious if we could get more specificity here.

Reply:
Thanks @cngonzalez will defer to @hermanwikner and @sjelfull on this.

Reply:
Also realize that this is rearranging pre-exsiting code so maybe better to address in another PR 😅

Reply:
I did some testing, and it appears we no longer need this check. The logic in this try/catch was initially added because the `/setup` endpoint used to return an error if the addon dataset had already been created. However, this no longer seems to be the case, as the endpoint now returns the name of the addon dataset if it has already been created instead – which is great!

Reply:
The comment below mentions that it could return an error also if the dataset doesn't exists. Maybe we can do a follow up to consider if we remove it or not, wouldn't feel confortable removing it in this one given it should be mostly focused on re-arranging the changes and not refactors. 
But also tried it and it doesn't fail if the dataset doesn't exists.